testing/supporting_docs/ui_selectors_and_their_importance.md

```diff
@@ -0,0 +1,178 @@
+
+The second way of adding an ‘accessibilityId’ to a UI element is to do so through Swift code.  
+This can be done by setting the ‘accessibilityIdentifier’ attribute of a UI element as shown 
+below.
+
+```swift
+if (indexPath.row == 0) {
+    fieldNameLabel.text = "Name"
+    fieldValueLabel.text = self.currentUser!.fullName
+    fieldValueLabel.accessibilityIdentifier = "username"
+} else {
+    fieldName.text = "Email"
+    fieldValueLabel.text = self.currentUser!.email
+    fieldValueLabel.accessibilityIdentifier = "email"
+}
+```
+
+### Android (AccessibilityId Example)
+Unlike iOS Android does not dynamically assign any values to the AccesibilityId.  Instead if 
+the developer wishes to use AccessibilityId for testing purposes, it can be added to Android 
+apps by setting the ‘contentDescription’ attribute either in the xml UI specifications or 
+through the use of code by using the ‘setContentDescription’ method of a UI element.  
+
+
+```xml
+<Button 
+    android:id="@+id/keycloakLogin"
+    android:contentDescription="@+string/authenticate_button"
+    style="@style/widget.AppCompat.Button.Colored"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:text="Authenticate"
+    android:layout_marginBottom="15dp"
+    android:layout_alignParentBottom="true"
+    android:layout_alignParentStart="true" />
+```
+
+
+![alt text](./images/authenticate_button_in_appium_inspector.png "Accessibility Id Now Available to Use")
+
+### Xamarin (AutomationId Example)
+Xamarin elements have a unique attribute called AutomationId, which can be used to set UI selectors 
+across both iOS and Android platforms.  This can be set in either the XAML code or in the C# backend 
+code.  
+
+```cs
+// C# Code
+var button = new Button{
+    Text = "Authenticate"
+    AutomationId = "authenticate_button"
+}
+```
+
+
+```xml
+<ic:CircleImage AutomationId="my_image" 
+    WidthRequest="100" 
+    HeightRequest="100"
+    Aspect="AspectFill" />
+```
+
+
+
+![img text](./images/accessibilityid_set_from_automationid.png "Resulting in AccessibilityId and Name Attributes Being set as Shown in Appium Inspector")
+
+### Cordova (id Example)
+
+Unlike Android, iOS and Cordova applications with can be tested using Appiums native context, webview
+context should be used to test cordova/ionic applications.  When testing applications using the webview
+context, web based targeting selectors should be used.  For example in the Java-client the full Selenium
+WebDriver API can be used, while in the Nodejs Webdriverio library the standard webdriverio selector 
+targeting methods can be used.  Note that in both of these libraries, the standard web based selector or
+locator strategies are not available when using the native context.
+
+In order to prepare Appium to test a cordova/ionic application in the desied capabilities section,
+`autoWebview` shoul be included and set to `true`.  This will immediatley prepare the test to run in a
```

Comment:
@twnolan 'shoul' needs to be replaced by 'should', and 'immediatley' should be 'immediately'